tutorials/advanced-topics/basic-segment-tree.md

```diff
@@ -0,0 +1,179 @@
+---
+
+title: 'Segment Tree'
+
+description: 'Author: @Dhruvil-Lakhtaria'
+
+hide_table_of_contents: true
+
+---
+
+<TutorialAuthors  names="@Dhruvil-Lakhtaria"/>  
+
+### Introduction
+  Segment Tree is a type of datastructure used in very specific kind of problems. Problems that deal with 2 types of operations namely point `update()` and `query()`(Some kind of queries like Max, Min, GCD, etc.) for a given array.
+
+### Structure
+It is basically a binary tree with 3 types of nodes - 
+
+ 1. `Root Node` - denotes the whole array.
+ 2. `Leaf Node` - denotes a single element of the array.
+ 3. `Internal Node` - denotes a sub-interval of the array.
+ 
+### Representation
+We can implement the above using an array representation.
+Each Index denotes the answer to that interval with $$(2*i+1) th$$ index as its left child and $$(2*i+2)th$$ index as its right child.
+
+![Segment-Tree](https://miro.medium.com/max/1100/1*9V1l8_weUfmT4jHg21EE5w.png)
```

Comment:
please update the diagram. it's hard to read with blue background and black text color